odoo/tools/json.py

```diff
@@ -66,7 +68,7 @@ def json_default(obj):
         return fields.Date.to_string(obj)
     if isinstance(obj, lazy):
         return obj._value
-    if isinstance(obj, ReadonlyDict):
+    if isinstance(obj, Mapping):
```

Comment:
If you change it, please align in xmlrpc.

Reply:
To remain consistent, I have modified the default for JSON serialisation which will check for `MappingProxyType`.